src/map.rs

```diff
@@ -4395,9 +4431,9 @@ impl<'a, 'b, K, Q: ?Sized, V, S, A: Allocator> VacantEntryRef<'a, 'b, K, Q, V, S
     /// }
     /// ```
     #[cfg_attr(feature = "inline-more", inline)]
-    pub fn insert_entry(self, value: V) -> OccupiedEntry<'a, K, V, S, A>
+    pub fn insert_entry(self, value: V) -> OccupiedEntry<'map, K, V, S, A>
     where
-        K: Hash + From<&'b Q>,
+        K: Hash + From<&'key Q>,
```

Comment:
Not _your_ problem, but this is an issue too, since downstream `Q` might implement a bad `From<&Q> for K`.

Reply:
yeah i noticed that too, that was half the reason i asked about the assert